### DISCUSSION TITLE
My new extension: Basic authentication support for Api2!

### DISCUSSION THREAD

**Discussion by elidrissidev** [OpenMage/magento-lts#1986] [Show & Tell]
Hello all,

As you all know Magento has only ever supported OAuth as an authentication method in Api2, which is not the easiest to use especially if you only want to quickly test something, but most importantly, it cannot be used to authenticate scripts or backends because it requires human interaction to click the "Authorize" button.

So after some thinking and a couple of months of part-time work, I finally finished my first Magento extension that is about supporting Basic authentication in Api2, it works by adding a new concept of "REST Users" that can be created and assigned a role, they can also be disabled anytime to stop their access.

The extension is completely free and available in my [GitHub repository](https://github.com/elidrissidev/Elidrissidev_Api2BasicAuth), it also comes with a thorough README and I'm willing to support it for the foreseeable future :).

Please feel free to report any issue if you do use it.

Have a great day!

**Comment by luigifab** [comment]
I suggest to add a `composer.json` to allow easy install/uninstall.

**Reply by elidrissidev** [reply]
I've never been aware of using composer to install extensions honestly. How does that work?

**Reply by luigifab** [reply]
For your module, that's easy, create a file like [this one](https://github.com/luigifab/openmage-apijs/blob/master/composer.json), and copy/paste your github repository here: https://packagist.org/packages/submit
For each releases, create a new tag.

To install your module (when OpenMage is managed by composer): `composer require your/extension [--ignore-platform-reqs]`

**Reply by elidrissidev** [reply]
Interesting. Thanks for sharing!

**Reply by kiatng** [reply]
@luigifab your [site](https://www.luigifab.fr/) can't be reached.

**Reply by luigifab** [reply]
🛰️  oh yes, you need an IPv6...
